src/index.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import ReactModal from 'react-modal';
+
+let rootSelector;
+
+export const setAppElement = (selector) => {
+    rootSelector = selector;
+    ReactModal.setAppElement(selector);
+};
+
+export class Modal extends Component {
+    state = {
+        isOpen: false,
```

Comment:
Can we opt-out of this behavior? Ideally, I would be able to control if the modal is open or not from the outside, which is useful in some scenarios (e.g.: if we have a modal manager somehow). Perhaps we could consider:

- If `isOpen` is passed, assume the component is uncontrolled (controlled by the parent)
- Otherwise, respect `initialIsOpen` and it's assumed the component is self-controlled

Thoughts?

Reply:
Initially I thought about supporting both use cases: controlled or uncontrolled. I actually did implement that, however I do think that it maybe a bit of a dangerous path to support both since we'll allow users to pass `isOpen` and override their own commands of open and close.

The only use case that it doesn't support is, as you said, the option to start the modal already open. But that can be implemented using `initialIsOpen` as you suggested.

However, I pushed the code I previously had in which I support both controlled and uncontrolled. Please comment what do you think.

I'm tending to delete that commit and add support for `initialIsOpen` but since I did implemented it on the beginning, I though I may as-well push that code.

Keeping this thread open.

Reply:
After some discussion, we decided to offer as a base package the Modal with only the iOS bug fixed and then another component that uses the Modal and controls it.